src/curated-corpus/components/CustomSectionFormConnector/CustomSectionFormConnector.tsx

```diff
@@ -14,72 +19,175 @@ interface CustomSectionFormConnectorProps {
+        ...(formValues.iabCategory && {
+          iab: {
+            taxonomy: 'IAB-3.0',
+            categories: [formValues.iabCategory],
+          },
+        }),
+      };
+
+      await runMutation(
+        updateSectionMutation,
+        {
+          variables: {
+            data: input,
+          },
+        },
+        'Custom section updated successfully',
+        (data) => {
+          if (onSuccess) {
+            onSuccess(existingSection.externalId);
+          }
         },
-      }),
-    };
+        () => {
+          // Error is already handled by runMutation
+        },
+      );
+    } else {
+      // Create new section
+      const input: CreateCustomSectionInput = {
+        title: formValues.title.trim(),
+        description: formValues.subtitle.trim(),
+        heroTitle: formValues.heroTitle?.trim() || undefined,
+        heroDescription: formValues.heroDescription?.trim() || undefined,
+        startDate: formValues.startDate!.toFormat('yyyy-MM-dd'),
+        endDate: formValues.endDate
+          ? formValues.endDate.toFormat('yyyy-MM-dd')
+          : undefined,
+        scheduledSurfaceGuid,
+        active: true,
+        disabled: false,
+        createSource: ActivitySource.Manual,
+        ...(formValues.iabCategory && {
+          iab: {
+            taxonomy: 'IAB-3.0',
+            categories: [formValues.iabCategory],
+          },
+        }),
+      };
 
-    await runMutation(
-      createSectionMutation,
-      {
-        variables: {
-          data: input,
+      await runMutation(
+        createSectionMutation,
+        {
+          variables: {
+            data: input,
+          },
         },
-      },
-      'Custom section created successfully',
-      (data) => {
-        const createdSection = data?.createCustomSection;
+        'Custom section created successfully',
+        (data) => {
+          const createdSection = data?.createCustomSection;
 
-        // Navigate to the section details page
-        if (createdSection?.externalId) {
-          history.push(
-            `/curated-corpus/custom-sections/${createdSection.externalId}/${scheduledSurfaceGuid}/`,
-          );
-        }
+          // Navigate to the section details page
+          if (createdSection?.externalId) {
+            history.push(
+              `/curated-corpus/custom-sections/${createdSection.externalId}/${scheduledSurfaceGuid}/`,
+            );
+          }
 
-        if (onSuccess) {
-          onSuccess(createdSection?.externalId);
-        }
-      },
-      () => {
-        // Error is already handled by runMutation
-      },
-    );
+          if (onSuccess) {
+            onSuccess(createdSection?.externalId);
+          }
+        },
+        () => {
+          // Error is already handled by runMutation
+        },
+      );
+    }
   };
 
+  const handleDelete = async () => {
+    if (!existingSection) return;
+
+    if (
+      window.confirm(
```

Comment:
instead of using the JS `window.confirm` method lets show our custom modal for alerting when deleting a custom section. It should be like th
<img width="960" height="483" alt="Screenshot 2025-09-30 at 19 17 51" src="https://github.com/user-attachments/assets/03c88451-0352-4eca-b3f5-222de2b6b150" />
e following modal, but no removal reasons & "OK" & "Cancel" buttons.